src/Runtime/Distributions/Automata/Automaton.GroupExtractor.cs

```diff
@@ -21,19 +21,19 @@ public abstract partial class Automaton<TSequence, TElement, TElementDistributio
     {
         private static class GroupExtractor
         {           
-            internal static Dictionary<byte, TThis> ExtractGroups(Automaton<TSequence, TElement, TElementDistribution, TSequenceManipulator, TThis> automaton)
+            internal static Dictionary<int, TThis> ExtractGroups(Automaton<TSequence, TElement, TElementDistribution, TSequenceManipulator, TThis> automaton)
             {
-                Dictionary<byte, HashSet<int>> subGraphs;
+                Dictionary<int, HashSet<int>> subGraphs;
                 var order = ComputeTopologicalOrderAndGroupSubgraphs(automaton, out subGraphs);
                 return BuildSubautomata(automaton.states, order, subGraphs);
             }
 
-            private static Dictionary<byte, TThis> BuildSubautomata(
+            private static Dictionary<int, TThis> BuildSubautomata(
```

Comment:
Would it make sense to go straight to `TThis[]` and `HashSet<int>[]` instead of `Dictionary<int, TThis>` and `Dictionary<int, HashSet<int>>`, respectively?